### PR TITLE
Geomesa 1503,1504: remove asm from compute jar, simplify scala 2.10 build

### PR DESCRIPTION
**Pull request by tkunicki**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1445,6 +1445,12 @@
                 <groupId>com.esotericsoftware.kryo</groupId>
                 <artifactId>kryo</artifactId>
                 <version>2.21</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>org.ow2.asm</groupId>
+                        <artifactId>asm</artifactId>
```
Comment (elahrvivaz) [inline pom.xml]: is asm still on the classpath at all, even in test scope? Seems like kryo might actually need asm, but maybe not in the way we use it.

Comment (tkunicki) [inline pom.xml]: Kyro does use asm, but it's shaded and relocated in the dependency we use.

**Comment by aheyne** [comment]
We might need to add the Scala 2.11 profile to the travis build command or set the Scala 2.11 profile to activate on the travis profile.

**Comment by tkunicki** [comment]
2.11 build is active by default

File: pom.xml (hunk 1 above)
Comment (elahrvivaz) [inline pom.xml]: nice, can we add a comment to the exclusion block so we remember that in the future?

**Comment by elahrvivaz** [comment]
looks good to me

**Comment by jnh5y** [comment]
@tkunicki two quick things:
Anthony likes commits to start with the line GEOMESA-####.  Note that GeoMesa is all caps and the dash is present.

Second, it looks like this has conflicts now...

**Comment by tkunicki** [comment]
@jnh5y rebased PR.  Don't understand your comments about my commit messages...?

**Pull request merged**